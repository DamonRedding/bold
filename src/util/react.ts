import { MutableRefObject, Ref } from 'react'

export function composeRefs<T>(...refs: Array<Ref<T>>) {
    return (value: T) => refs.forEach(ref => setRef(ref, value))
}

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}

/**
 * Returns the next sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getNextSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.nextElementSibling) {
        current = current.nextElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}

/**
 * Returns the previous sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getPreviousSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.previousElementSibling) {
        current = current.previousElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}
