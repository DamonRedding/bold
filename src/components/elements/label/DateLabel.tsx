import * as React from 'react'
import { ComposedLabelProps, ComposedLabel } from './ComposedLabel'
import DateUtil from '../../../util/DateUtil'

export interface DateLabelProps extends ComposedLabelProps {
    value: string
}

export class DateLabel extends React.Component<DateLabelProps, any> {

    render() {
        const { value, ...rest } = this.props
        return (
            <ComposedLabel {...rest} >
                {value && DateUtil.asClientFormat(value)}
            </ComposedLabel>
        )
    }

}
