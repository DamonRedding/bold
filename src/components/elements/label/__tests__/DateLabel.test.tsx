import * as React from 'react'
import { shallowRenderAndMatch } from '../../../../__tests__/test.defaults'
import { DateLabel } from '../DateLabel'

describe('DateLabel', () => {
    it('render', () => {
        shallowRenderAndMatch(
            <DateLabel title='label' value='1990-10-10' />
        )
    })
    it('size small', () => {
        shallowRenderAndMatch(
            <DateLabel title='label' value='1990-10-10' size='small' />
        )
    })
    it('horizontal', () => {
        shallowRenderAndMatch(
            <DateLabel title='label' value='1990-10-10' horizontal />
        )
    })
    it('horizontal e small', () => {
        shallowRenderAndMatch(
            <DateLabel title='label' value='1990-10-10' horizontal size='small' />
        )
    })
})
