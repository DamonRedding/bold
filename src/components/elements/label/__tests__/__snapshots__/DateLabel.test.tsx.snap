// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateLabel horizontal 1`] = `
<ComposedLabel
  horizontal={true}
  placeholder="N達o informada"
  title="label"
>
  10/10/1990
</ComposedLabel>
`;

exports[`DateLabel horizontal e small 1`] = `
<ComposedLabel
  horizontal={true}
  placeholder="N達o informada"
  size="small"
  title="label"
>
  10/10/1990
</ComposedLabel>
`;

exports[`DateLabel render 1`] = `
<ComposedLabel
  placeholder="N達o informada"
  title="label"
>
  10/10/1990
</ComposedLabel>
`;

exports[`DateLabel size small 1`] = `
<ComposedLabel
  placeholder="N達o informada"
  size="small"
  title="label"
>
  10/10/1990
</ComposedLabel>
`;
